src/functests/test_homepage.py

```diff
@@ -18,52 +20,113 @@ def setUp(self):
     def tearDown(self):
         self.browser.quit()
 
-    def test_in_homepage_should_see_default_links(self):
+    def _login(self, email, password):
+        login_input_elt = self.browser.find_element_by_name('email')
+        login_input_elt.send_keys(email)
+        password_input_elt = self.browser.find_element_by_name('password')
+        password_input_elt.send_keys(password)
+        login_button = self.browser.find_element_by_name('login-button')
+        login_button.click()
+
+    def _register(self, email, password, interests):
+        login_input_elt = self.browser.find_element_by_name('email')
+        login_input_elt.send_keys(email)
+        password_input_elt = self.browser.find_element_by_name('password')
+        password_input_elt.send_keys(password)
+        interests_input_elt = self.browser.find_element_by_name('interests')
+        interests_input_elt.send_keys(interests)
+        register_button = self.browser.find_element_by_name('register-button')
+        register_button.click()
+
+    def test_login_with_unknown_email(self):
+        self.browser.get('http://localhost:8080')
+
+        self._login('unknown@email.com', 'unknownpassword')
+
+        error_message = self.browser.find_element_by_name('error-message').text
+        self.assertEquals('Unknown email or invalid password', error_message)
+
+    def test_login_with_invalid_password(self):
+        daltesthelpers.init_user_dummy('known_user@gator.com', '', '')
+
+        self.browser.get('http://localhost:8080')
+
+        self._login('known_user@gator.com', 'invalid_password')
+
+        error_message = self.browser.find_element_by_name('error-message').text
+        self.assertEquals('Unknown email or invalid password', error_message)
+
+    def test_register(self):
+        self.browser.get('http://localhost:8080/register')
+
+        self._register('register_then_login@gator.com', 'register_then_login', 'interests')
+
+        self.assertIsNotNone(dal.get_user('register_then_login@gator.com'))
+
```

Comment:
At the end of the test, perhaps we shoud also check the user has been properly redirected, not only that he is in database